src/notebook/model.ts

```diff
@@ -299,175 +199,96 @@ class NotebookModel implements INotebookModel {
-      return;
-    }
-    let oldValue = this._defaultMimetype;
-    this._defaultMimetype = newValue;
-    let name = 'defaultMimetype';
-    this.stateChanged.emit({ name, oldValue, newValue });
-  }
-
-  /**
-   * The read-only status of the notebook.
+   * The major version number of the nbformat.
+   *
+   * #### Notes
+   * This is a read-only property.
    */
-  get readOnly(): boolean {
-    return this._readOnly;
-  }
-  set readOnly(newValue: boolean) {
-    if (newValue === this._readOnly) {
-      return;
-    }
-    let oldValue = this._readOnly;
-    this._readOnly = newValue;
-    let cells = this._cells;
-    for (let i = 0; i < cells.length; i++) {
-      cells.get(i).input.textEditor.readOnly = newValue;
-    }
-    let name = 'readOnly';
-    this.stateChanged.emit({ name, oldValue, newValue });
+  get nbformat(): number {
+    return this._nbformat;
   }
 
   /**
-   * The session for the notebook.
+   * The minor version number of the nbformat.
+   *
+   * #### Notes
+   * This is a read-only property.
    */
-  get session(): INotebookSession {
-    return this._session;
-  }
-  set session(newValue: INotebookSession) {
-    if (newValue === this._session) {
-      return;
-    }
-    let oldValue = this._session;
-    this._session = newValue;
-    NotebookModelPrivate.sessionChanged(this, newValue);
-    let name = 'session';
-    this.stateChanged.emit({ name, oldValue, newValue });
+  get nbformatMinor(): number {
+    return this._nbformatMinor;
   }
 
   /**
-   * The kernelspec metadata for the notebook.
+   * The dirty state of the model.
+   *
+   * #### Notes
+   * This should be cleared when the document is loaded from
+   * or saved to disk.
    */
-  get kernelspec(): IKernelspecMetadata {
-    return JSON.parse(this._kernelspec);
+  get dirty(): boolean {
+    return this._dirty;
   }
-  set kernelspec(newValue: IKernelspecMetadata) {
-    let oldValue = JSON.parse(this._kernelspec);
-    if (shallowEquals(oldValue, newValue)) {
+  set dirty(value: boolean) {
+    if (value === this._dirty) {
       return;
     }
-    this._kernelspec = JSON.stringify(newValue);
-    let name = 'kernelspec';
-    this.stateChanged.emit({ name, oldValue, newValue });
+    this._dirty = value;
+    this.dirtyChanged.emit(value);
   }
 
   /**
-   * The language info metadata for the notebook.
+   * The read-only state of the model.
    */
-  get languageInfo(): ILanguageInfoMetadata {
-    return JSON.parse(this._langInfo);
+  get readOnly(): boolean {
+    return this._readOnly;
   }
-  set languageInfo(newValue: ILanguageInfoMetadata) {
-    let oldValue = JSON.parse(this._langInfo);
-    if (shallowEquals(oldValue, newValue)) {
+  set readOnly(value: boolean) {
+    if (value === this._readOnly) {
       return;
     }
-    this._langInfo = JSON.stringify(newValue);
-    let name = 'languageInfo';
-    this.stateChanged.emit({ name, oldValue, newValue });
+    this._readOnly = value;
```

Comment:
We should emit some sort of signal here - I imagine a widget would really like to know if a model becomes read only or not.